internal/background/availability_queue_test.go

```diff
@@ -60,22 +64,22 @@ func TestFullQueueSend(t *testing.T) {
 
 func TestQueueCancelSend(t *testing.T) {
 	ctx := context.Background()
+	ctx = identity.WithIdentity(t, ctx)
 	_ = kafka.InitializeStubBroker(16)
 
 	// enqueue message to be sent first
-	msg, _ := kafka.AvailabilityStatusMessage{SourceID: "1"}.GenericMessage()
+	msg, _ := kafka.AvailabilityStatusMessage{SourceID: "1"}.GenericMessage(ctx)
 	EnqueueAvailabilityStatusRequest(&msg)
-
+	// set the receiving message function up
+	wg := sync.WaitGroup{}
+	wg.Add(2)
 	// start sending messages
 	senderCtx, senderCancel := context.WithCancel(ctx)
 	go sendAvailabilityRequestMessages(senderCtx, 2, 5*time.Second)
 
 	// allow the other goroutine to put the message into the buffer
 	runtime.Gosched()
 
-	// set the receiving message function up
-	wg := sync.WaitGroup{}
-	wg.Add(1)
 	consumeCtx, consumeCancel := context.WithCancel(ctx)
 	defer consumeCancel()
```

Comment:
I do not understand this hunk, can you comment what was the intention? The way I read this is that this test should get stuck waiting forever, you add 2 to the waitgroup but Done is called just once so Wait will block forever.

Reply:
I got an error of negative counter for the wait group. 
After looking at other tests I saw that we are missing one because we have a goroutine for the **Consume** function. 
when I changed the wait group to two the error was solved.